Azure.Data.Wrappers.Test/Integration/TableStorageTests.cs

```diff
@@ -761,7 +762,42 @@ public void Dispose()
                 Assert.IsTrue(exists);
             }
         }
+        private class TestRetryPolicy : IRetryPolicy
+        {
+            public IRetryPolicy CreateInstance()
+            {
+                throw new NotImplementedException();
+            }
 
+            public bool ShouldRetry(int currentRetryCount, int statusCode, Exception lastException, out TimeSpan retryInterval, OperationContext operationContext)
+            {
+                throw new NotImplementedException();
+            }
+        }
+        [Test]
+        public void QueryWithOptions()
+        {
+            TableRequestOptions options = new TableRequestOptions();
+            options.RetryPolicy = new TestRetryPolicy();
```

Comment:
Make this a mock of IRetryPolicy and have your assert validate that the retry policy is called at least once.